packages/engine/Source/Core/addAll.js

```diff
@@ -0,0 +1,41 @@
+import defined from "./defined.js";
+
+/**
+ * Adds all elements from the given source array to the given target array.
+ *
+ * If the `source` is `undefined`, then nothing will be done. Otherwise,
+ * this has the same semantics as
+ * ```
+ * for (const s of source) target.push(s);
+ * ```
+ * but is usually more efficient than a `for`-loop, and does not put the
+ * elements of the source on the stack, as it would be done with the
+ * spread operator or when using `target.push.apply(source)`.
+ *
+ * @function
+ * @private
+ *
+ * @param {Array|undefined} source The source array
+ * @param {Array} target The target array
+ *
+ * @example
+ * const target = [ 0, 1, 2 ];
+ * const source = [ 3, 4, 5 ];
+ * Cesium.addAll(source, target);
+ * // The target is now [ 0, 1, 2, 3, 4, 5 ]
+ */
+function addAll(source, target) {
```

Comment:
I don't think I like the name `addAll`. I don't feel it's descriptive enough to understand what it does without looking at the documentation or the function itself. `addAllToArray` or maybe `combineArrays` could be better? Or, given my previous [comment](https://github.com/CesiumGS/cesium/pull/12176#issuecomment-2379407987) on this topic with this essentially doing what [`Array.concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) does, maybe `concatInto` or `concatIntoArray` would be more accurate?

Reply:
No _really_ strong opinions, just things that I thought about:

- In the linked (soon to be obsolete) PR, I initially called that function `pushAll`, but I wanted to emphasize that it does exactly not `push`...
- Any name that involves `concat` could easily be confused with `Array.concat`, and (as discussed in the other PR), that has a completely different semantic: People might expect it to return a concatenation. I think it's important to make clear that it operates _in-place_, and _on_ the given target array
- `combine...` could be OK (_maybe_ people would expect it to return something there, but not necessarily)

Maybe something like `appendAll` or `append(All?To?)Array` or so ...?

Reply:
Changed to `addAllToArray` - sounds reasonable and is long and clear enough for a global function.